src/pages/sighting/encounters/Encounters.jsx

```diff
@@ -85,26 +97,59 @@ export default function Encounters({
   const encounterFieldSchemas = useEncounterFieldSchemas();
   const encounters = get(sightingData, 'encounters', []);
 
+  useEffect(
+    () => {
+      setMessageForConfirmDelete(
+        'CONFIRM_DELETE_ENCOUNTER_DESCRIPTION',
+      );
+      if (vulnerableObject) {
+        setMessageForConfirmDelete('BOTH_VULNERABLE_MESSAGE');
+      }
+    },
+    [vulnerableObject],
+  );
+
   return (
     <div>
       <ConfirmDelete
         open={Boolean(encounterToDelete)}
         onClose={() => setEncounterToDelete(null)}
         onDelete={async () => {
           let successful;
+          const vulnObjKeys = vulnerableObject
+            ? Object.keys(vulnerableObject)
+            : [];
+          const vulnerableObjStr = get(vulnObjKeys, '0');
           if (pending) {
             successful = await deleteAGSEncounter(
               sightingId,
               encounterToDelete,
             );
-          } else {
+          }
+          if (!pending && vulnerableObjStr) {
```

Comment:
what's the point of `vulnerableObjStr`? unless i'm reading wrong you could just check `if (!pending && vulnerableObject)`

Reply:
I feel like I tried this, and the truthy wasn't doing what I expected. Trying again now.

Reply:
Works fine now. I wonder what I was doing last time lol 🤔 .
Done.